cli/src/api/mod.rs

```diff
@@ -573,6 +589,66 @@ impl PhylumApi {
     }
 }
 
+/// Phylum group types.
+pub enum Group {
+    Legacy(String),
+    Org(OrgGroup),
+}
+
+impl Group {
+    /// Create a new group.
+    fn new(org: String, group: String) -> Self {
+        Self::Org(OrgGroup { org, name: group })
+    }
+
+    /// Create a group without organization.
+    fn new_legacy(group: String) -> Self {
+        Self::Legacy(group)
+    }
+
+    /// Create a group from an optional group name.
+    pub fn try_new<O, G>(org: Option<O>, group: Option<G>) -> Option<Self>
+    where
+        O: Into<String>,
+        G: Into<String>,
+    {
+        group.map(|group| match org {
+            Some(org) => Self::new(org.into(), group.into()),
+            None => Self::new_legacy(group.into()),
+        })
+    }
+
+    /// Get the group's organization.
+    pub fn org(&self) -> Option<&str> {
+        match self {
+            Self::Legacy(_) => None,
+            Self::Org(OrgGroup { org, .. }) => Some(org.as_str()),
+        }
+    }
+
+    /// Get the group's name.
+    pub fn name(&self) -> &str {
+        match self {
+            Self::Legacy(name) => name.as_str(),
+            Self::Org(OrgGroup { name, .. }) => name.as_str(),
+        }
+    }
+
+    /// Format the group as a single string.
+    pub fn combined_format(&self) -> Cow<'_, str> {
+        match self {
+            Self::Legacy(name) => Cow::Borrowed(name),
+            Self::Org(OrgGroup { org, name }) => Cow::Owned(format!("{org}/{name}")),
+        }
+    }
```

Comment:
Here the group name is the group name and the group path is the combined format, but in functions like `create_project` the "combined format" value is used as the group name.

Reply:
Yes, because those are API functions and `group_name` is what the API calls it for now. That behavior is not exposed outside of those functions at all so I don't see why it would be an issue?

Reply:
The code is difficult to follow because we use the same few variable names, "group" "group name" "org group," in different places to refer to the name of the group, the organization/group path of the group (which may be just the group name for legacy groups), and a struct containing the name of the organization and group names. They're all overloaded names and they're all roughly the same concept so context doesn't help.

Reply:
That is correct, but that is also largely due to the way it was initially implemented and it will change in the future.

Generally I think calling this combined format `org_group` while `group` is just the group and `org` is just the org makes sense and that's what I've tried to use in the places this new struct has been used so far.